bin/sync.sh

```diff
@@ -28,49 +28,24 @@ if [ -n "$chart_name" ] && [ -d "$SCRIPT_DIR/../charts/$chart_name" ]; then
 fi
 
 # install s3 plugin
-# At the time of writing, version 0.7.0 was installed.
-# Hopefully version locking isn't necessary
-helm plugin list | grep "s3" > /dev/null || helm plugin install https://github.com/hypnoglow/helm-s3.git
-
-# Note: on providing a public URL for charts synced to S3
-# * This is not yet supported by helm-s3 plugin until
-#   https://github.com/hypnoglow/helm-s3/pull/56 is implemented
-# * This workaround uses two folders on s3;
-#   - one to sync with helm-s3,
-#   - a second one with a manually-changed index.yaml to allow fetching charts from public URLs
-
-INDEX_S3_DIR="charts-tmp"
-PUBLIC_DIR="charts"
-workaround_issue_helm_s3_56() {
-    # retrieve index
-    aws s3 cp s3://public.wire.com/$INDEX_S3_DIR/index.yaml index.yaml
-
-    # sync from $INDEX_S3_DIR to charts directory
-    if [ -n "$chart_name" ]; then
-        # Read chart urls into a bash array
-        # mapfile/readarray are nicer, but don't work on Mac's builtin bash
-        urls=()
-        while IFS='' read -r line; do array+=("$line"); done < <(yq r index.yaml 'entries.*.*.urls.0' | awk -F '- ' '{print $2$3}' | grep "$chart_name")
-    else
-        urls=()
-        while IFS='' read -r line; do array+=("$line"); done < <(yq r index.yaml 'entries.*.*.urls.0' | awk -F '- ' '{print $2$3}')
-    fi
-    for url in "${urls[@]}"; do
-        newurl=${url/$INDEX_S3_DIR/$PUBLIC_DIR};
-        echo "old=$url and new=$newurl"
-        aws s3 cp "$url" "$newurl"
-    done
-
-    # update and upload index to charts
-    sed -i'.bak' -e 's/s3\:\/\//https\:\/\/s3-eu-west-1.amazonaws.com\//g' index.yaml
-    aws s3 cp index.yaml s3://public.wire.com/$PUBLIC_DIR/index.yaml
-}
+# See https://github.com/hypnoglow/helm-s3/pull/56 for reason to use fork
+s3_plugin_version=$(helm plugin list | grep "^s3 " | awk '{print $2}')
+if [[ $s3_plugin_version != "0.9.0" ]]; then
+    echo "not version 0.9.0 from steven-sheehy fork, upgrading or installing plugin..."
```

Comment:
isn't this a little fragile? what happens if upstream releases a 0.9.0 that does not have this feature?

Reply:
Fragility fixed in last commit, where I also check the git hash.